src/AppBundle/Controller/MainController.php

```diff
@@ -0,0 +1,210 @@
+
+        $em = $this->getDoctrine()->getManager();
+
+        $em->persist($todo);
+
+        $em->flush();
+        
+        return new JsonResponse([ 'id' => $todo->getId(), 'body' => $todo->getBody(), ]);
+    }
+
+
+    /**
+     * @Route("/main/show", name="show_todos", options={"expose"=true})
+     */
+    public function showProductsAction()
+    {
+        $user = $this->getUser();
+
+        $todos = $user->getTodos();
+
+        
+        $todos = $todos->toArray();
+        $todosArray = [];
+        foreach ($todos as $todo) {
+            $todosArray[] = ['id' => $todo->getId(), 'body' => $todo->getBody(), 'completed' => $todo->getCompleted()];
+        }
+
+        return new JsonResponse($todosArray);
+    }
+
+    /**
+     * @Route("/main/remove", name="remove_todo", options={"expose"=true})
+     */
+    public function removeAction(Request $request)
+    {   
+        $user = $this->getUser();
+        $em = $this->getDoctrine()->getManager();
+
+        $todoId = $request->request->get('id');
+        $todo = $em->getRepository('AppBundle:Todo')->find($todoId);
+
+        if (!$todo) {
+            throw $this->createNotFoundException(
+                'No product found for id '.$todoId
+            );
+        }
+
+        $user->removeTodo($todo);
+        $em->remove($todo);
+        $em->flush();
+
+
+        return new JsonResponse("Item removed Correctly");
+    }
+
+    /**
+     * @Route("/main/complete", name="complete_todo", options={"expose"=true})
+     */
+    public function completeAction(Request $request)
+    {   
+        $user = $this->getUser();
+        $em = $this->getDoctrine()->getManager();
+
+        $todoId = $request->request->get('id');
+        $todo = $em->getRepository('AppBundle:Todo')->find($todoId);
+
+        if (!$todo) {
+            throw $this->createNotFoundException(
+                'No product found for id '.$todoId
+            );
+        }
+
+        $todo->setCompleted(! $todo->getCompleted());
+        $em->flush();
+
+
+        return new JsonResponse("Completed state updated with".$todo->getCompleted());
+    }
+
+
+    /**
+     * @Route("/main/completeAll", name="complete_all", options={"expose"=true})
+     */
+    public function completeAllAction()
+    {
+
+        $em = $this->getDoctrine()->getManager();
+
+        $user = $this->getUser();
+        $todos = $user->getTodos();
+        $todos = $todos->toArray();
+
+        $actionDone = '';
+        $falseItemsNumber = 0;
+        foreach ($todos as $todo) {
+            if (! $todo->getCompleted()){
+                $falseItemsNumber++;
+            }
+        }
+
+        if ($falseItemsNumber == 0){
+
+            foreach ($todos as $todo) {
+                $todo->setCompleted(false);
+            }
+            $actionDone = 'makeAllFalse';
+
+        } else {
+
+            foreach ($todos as $todo) {
+                $todo->setCompleted(true);
+            }
+            $actionDone = 'makeAllTrue';
```

Comment:
Again, such an interesting way of looking at things! I'm loving this new logic!!!